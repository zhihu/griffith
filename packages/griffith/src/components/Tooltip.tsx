import React, {Component} from 'react'
import {css} from 'aphrodite/no-important'
import {LocaleConfigKey} from '../constants/locales'
import TranslatedText from './TranslatedText'
import styles from './Tooltip.styles'

type Props = {
  content: LocaleConfigKey
}

class Tooltip extends Component<Props> {
  render() {
    const {content} = this.props
    return (
      <div className={css(styles.root)}>
        <TranslatedText name={content} />
      </div>
    )
  }
}

export default Tooltip
