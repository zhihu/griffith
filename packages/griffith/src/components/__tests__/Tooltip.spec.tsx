import React from 'react'
import Renderer from 'react-test-renderer'
import Tooltip from '../Tooltip'

describe('Tooltip', () => {
  it('get Tooltip component', () => {
    expect(
      Renderer.create(<Tooltip content="quality-hd" />).toJSON()
    ).toMatchSnapshot()
  })
})
