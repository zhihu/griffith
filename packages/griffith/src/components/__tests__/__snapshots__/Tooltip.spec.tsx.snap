// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip get Tooltip component 1`] = `
<div
  className="root_r31oyg"
>
  HD
</div>
`;
